API Keys content update (#3594)
temporalio/documentation: docs/develop/java/temporal-client.mdx

## Connect to Temporal Cloud {#connect-to-temporal-cloud}

### How to connect a Temporal Client to Temporal Cloud using an API key {#connect-to-temporal-cloud-api-key}

To use an [API key](/cloud/api-keys) with the Temporal Jave SDK, use the following code example.

```java
    WorkflowServiceStubs service =
        WorkflowServiceStubs.newServiceStubs(
            WorkflowServiceStubsOptions.newBuilder()
                .addApiKey(
                    () ->
                        <API key>)
                .setTarget(<endpoint>)
                .setEnableHttps(true)
                ...
                .build());

    WorkflowClient client =
        WorkflowClient.newInstance(

>             service, WorkflowClientOptions.newBuilder().setNamespace(<namespace_id>.<account_id>).build());

Reviewer: ```suggestion
            service, WorkflowClientOptions.newBuilder().setNamespace(<namespace_id>.<account_id>).build());
            ```
```

> // Update by replacing, this must be done in a thread safe way

Reviewer 2: Can we clarify what "thread safe way" means specifically in this context?

> // Update by replacing, this must be done in a thread safe way
> myKey = "Bearer " + <new APIKey>;
> ```

Reviewer: ```suggestion
To update an API key:
```java
String myKey = <APIKey>;
WorkflowServiceStubsOptions stubOptions =
    WorkflowServiceStubsOptions.newBuilder()
        .addApiKey(() -> myKey)
        .build();
// Update by replacing, this must be done in a thread safe way
myKey = "Bearer " + <new APIKey>;
```
```

## Connect to Temporal Cloud {#connect-to-temporal-cloud}

### How to connect a Temporal Client to Temporal Cloud using an API key

> To use your [API key](/cloud/api-keys) with a Temporal SDK, include the API key in the Bearer token header and specify the Namespace header. The Bearer token header indicates the authentication type, while the Namespace header routes the request.

Reviewer: ```suggestion
To use an [API key](/cloud/api-keys) with Temporal Java SDK, leverage the following code sample.
```

> At a client level:
> 
> ```java
> // Create the Workflow service stub.
> WorkflowServiceStubsOptions.Builder stubOptions =
>     WorkflowServiceStubsOptions.newBuilder()
>         .addApiKey(() -> <APIKey>)
>         .setEnableHttps(true)
>         .setTarget(<endpoint>);
> 
> WorkflowServiceStubs service = WorkflowServiceStubs.newServiceStubs(stubOptions.build());
> 
> /*
>  * Get a Workflow service client which can be used to start, Signal, and Query Workflow Executions.
>  */
> WorkflowClient client =
>     WorkflowClient.newInstance(
>         service, WorkflowClientOptions.newBuilder().setNamespace(<namespace_id>.<account_id>).build());
> ```
> 
> At a per-call level:
> 
> ```java
> String myKey = <APIKey>;
> WorkflowServiceStubsOptions stubOptions =
>     WorkflowServiceStubsOptions.newBuilder()
>         .addApiKey(() -> myKey)
>         .build();
> 

Reviewer: ```suggestion
    WorkflowServiceStubs service =
        WorkflowServiceStubs.newServiceStubs(
            WorkflowServiceStubsOptions.newBuilder()
                .addApiKey(
                    () ->
                        <API key>)
                .setTarget(<endpoint>)
                .setEnableHttps(true)
                ...
                .build());

    WorkflowClient client =
        WorkflowClient.newInstance(
            service, WorkflowClientOptions.newBuilder().setNamespace(<namespace_id>.<account_id>).build());
```

> // Just update by replacing, this must be done in a thread safe way

Reviewer: ```suggestion
// Update by replacing, this must be done in a thread safe way
```
